src/components/Sidebar/index.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { KeyCodes } from '../../utils/keyCodes';
+import { SidebarPopup } from './SidebarPopup.part';
+import { SidebarActivatorContainer } from './SidebarActivatorContainer.part';
+
+type PositionType = 'top' | 'left' | 'bottom' | 'right';
+
+export interface SidebarProps {
+  /**
+   * Element which triggers sidebar showing by click.
+   */
+  activator: React.ReactChild;
+  /**
+   * Determines where to show a sidebar. Valid positions are: left, right, top, bottom.
+   * @default right
+   */
+  position?: PositionType;
+  /**
+   * Size in valid css units.
+   * @default 275px
+   */
+  size?: string;
+}
+
+interface SidebarState {
+  open: boolean;
+}
+
+export class Sidebar extends React.Component<SidebarProps, SidebarState> {
+  constructor(props: SidebarProps) {
+    super(props);
+    this.state = {
+      open: false,
```

Comment:
Why don't use the provided value from props like `open: open === true`? It might be useful.